src/mam4xx/mo_gas_phase_chemdr.hpp

```diff
@@ -388,14 +388,54 @@ void perform_atmospheric_chemistry_and_microphysics(
     }
     // do aerosol microphysics (gas-aerosol exchange, nucleation,
     // coagulation)
+    Real gas_aero_exchange_condensation[gas_pcnst] = {};
+    Real gas_aero_exchange_renaming[gas_pcnst] = {};
+    Real gas_aero_exchange_nucleation[gas_pcnst] = {};
+    Real gas_aero_exchange_coagulation[gas_pcnst] = {};
+    Real gas_aero_exchange_renaming_cw[gas_pcnst] = {};
     mam4::microphysics::modal_aero_amicphys_intr(
         // in
         config_amicphys, dt, temp, pmid, pdel, zm, pblh, qv, cldfrac,
         // out
         vmr, vmrcw,
+        // diagnostics (out)
+        gas_aero_exchange_condensation, gas_aero_exchange_renaming,
+        gas_aero_exchange_nucleation, gas_aero_exchange_coagulation,
+        gas_aero_exchange_renaming_cw,
         // in
         vmr0, vmr_pregas, vmr_precld, dgncur_a_kk, dgncur_awet_kk, wetdens_kk);
 
+    if (diag_arrays.gas_aero_exchange_condensation.size()) {
+      for (int i = 0; i < gas_pcnst; ++i) {
+        diag_arrays.gas_aero_exchange_condensation(i, kk) =
+            gas_aero_exchange_condensation[i];
+      }
+    }
+    if (diag_arrays.gas_aero_exchange_renaming.size()) {
+      for (int i = 0; i < gas_pcnst; ++i) {
+        diag_arrays.gas_aero_exchange_renaming(i, kk) =
+            gas_aero_exchange_renaming[i];
+      }
+    }
+    if (diag_arrays.gas_aero_exchange_nucleation.size()) {
+      for (int i = 0; i < gas_pcnst; ++i) {
+        diag_arrays.gas_aero_exchange_nucleation(i, kk) =
+            gas_aero_exchange_nucleation[i];
+      }
+    }
+    if (diag_arrays.gas_aero_exchange_coagulation.size()) {
+      for (int i = 0; i < gas_pcnst; ++i) {
+        diag_arrays.gas_aero_exchange_coagulation(i, kk) =
+            gas_aero_exchange_coagulation[i];
+      }
+    }
+    if (diag_arrays.gas_aero_exchange_renaming_cw.size()) {
+      for (int i = 0; i < gas_pcnst; ++i) {
+        diag_arrays.gas_aero_exchange_renaming_cw(i, kk) =
+            gas_aero_exchange_renaming_cw[i];
+      }
+    }
```

Comment:
I am not sure if it makes sense to convert this for-loop to a parallel_for or not. Also, we can create a function for the parallel_for as it is repeated multiple times. Something like:
```
if (diag_arrays.gas_aero_exchange_coagulation.size()) 
   populate_diags(gas_aero_exchange_coagulation, diag_arrays.gas_aero_exchange_coagulation)
```
We can do this in a follow-up PR as well.

Reply:
We can consider it in a follow-up PR.

Once you are ready, let me know, and I can run this on Frontier before we start merging this.